web-console/src/components/segment-timeline.tsx

```diff
@@ -0,0 +1,393 @@
+    if (loading) {
+      return <div>
+        <Loader
+          loading={loading}
+        />
+      </div>;
+    }
+
+    if (error) {
+      return <div>
+        <span className={'no-data-text'}>Error when loading data: {error}</span>
+      </div>;
+    }
+
+    if (xScale === null || yScale === null) {
+      return <div>
+        <span className={'no-data-text'}>Error when calculating scales</span>
+      </div>;
+    }
+
+    if (activeDatasource !== null && data[activeDataType].every((d: any) => d[activeDatasource] === undefined)) {
+      return <div>
+        <span className={'no-data-text'}>No data available for <i>{activeDatasource}</i></span>
+      </div>;
+    }
+
+    const barCounts = data[activeDataType].length;
+    const barWidth = (chartWidth - this.chartMargin.left - this.chartMargin.right) / barCounts;
+    return <StackedBarChart
+      dataToRender={dataToRender}
+      svgHeight={chartHeight}
+      svgWidth={chartWidth}
+      margin={this.chartMargin}
+      changeActiveDatasource={(e: string) => this.setState({activeDatasource: e})}
+      activeDataType={activeDataType}
+      formatTick={(n: number) => this.formatTick(n)}
+      xScale={xScale}
+      yScale={yScale}
+      barWidth={barWidth}
+    />;
+  }
+
+  render() {
+    const { datasources,  activeDataType, activeDatasource, timeSpan } = this.state;
+
+    return <div className={'segment-timeline app-view'}>
+      {this.renderStackedBarChart()}
+      <div className={'side-control'}>
+        <FormGroup>
+          <RadioGroup
+            onChange={(e: any) => this.setState({activeDataType: e.target.value})}
+            selectedValue={activeDataType}
+          >
+            <Radio label={'Segment count'} value={'countData'}/>
+            <Radio label={'Total size'} value={'sizeData'}/>
+          </RadioGroup>
+        </FormGroup>
+
+        <FormGroup
+          label={'Datasource:'}
+        >
+          <HTMLSelect
+            onChange={(e: any) => this.setState({activeDatasource: e.target.value === 'all' ? null : e.target.value})}
+            value={activeDatasource == null ? 'all' : activeDatasource}
+            fill
+          >
+            <option value={'all'}>Show all</option>
+            {
+              datasources.map(d => {
+                return <option key={d} value={d}>
+                  {d}
+                </option>;
+              })
+            }
+          </HTMLSelect>
+        </FormGroup>
+
+        <FormGroup
+          label={'Period:'}
+        >
+          <HTMLSelect
+            onChange={(e: any) => this.onTimeSpanChange(e.target.value)}
+            value={timeSpan}
+            fill
+          >
+            <option value={1}> 1 months</option>
+            <option value={3}> 3 months</option>
+            <option value={6}> 6 months</option>
+            <option value={9}> 9 months</option>
+            <option value={12}> 1 year</option>
```

Comment:
What if I have more than a year of segments? Any reason not to have an 'all' option, or alternatively an input of some sort for a start or end anchor date for these options?

Reply:
Currently, it supports a fixed period of time (mostly for demo purpose, I chose the number of months for no reason), but I could easily add a custom option to allow the users to select their own choices of timespan or start/end date